spec/PeriodicBackgroundSync-index.bs

```diff
@@ -145,6 +148,30 @@ Fetch requests within the [=periodicsync event=] while [=periodicsync event/in t
 
+Security Considerations {#security}
+====================================
+Since Periodic Background Sync is a [=service worker=] based API, [[service-workers#security-considerations]] apply.
+
+In addition, if there is a disparity between the intervals at which the web app can synchronize with its servers and those at which the user visits the site, the [=origin=] gets more power than the user is aware of or intends. This is because the [=origin=] is able to execute code more frequently than the user intends, from within the [=periodicsync event=] while [=periodicsync event/in the background=].
+User agents should therefore implement countermeasures to prevent such disparity.
+
+To mitigate the above, the user agent MUST:
+* Provide UI for users to opt out of firing [=periodicsync events|periodicsync event=] for any given [=origin=].
+* Impose a time limit on the lifetime extension and execution time of a {{PeriodicSyncEvent}} which is stricter than the [=ExtendableEvent/timed out flag|time limit=] imposed for {{ExtendableEvent}}s in general. In particular, any retries of the {{PeriodicSyncEvent}} MAY have a significantly shortened time limit.
+
+  Note: {{PeriodicSyncManager/register(tag, options)}} is blocked if invoked for a [=/service worker registration=], where none of the [=service worker clients=] for the associated [=origin=] have a [=frame type=] of "`top-level`" or "`auxiliary`".
+
+The user agent MUST also impement at least one of the following:
+* Auto-deny the capability to an [=origin=] for which the user has not expressed a level of trust.
+
+  Note: This trust may be expressed by installing the website as a web app.
+
+* Cap the interval at which [=periodicsync event|periodicsync events=] are fired. This can be done by setting [=minimum periodic sync interval for any origin=] and [=minimum periodic sync interval across origins=] as defined in [[#constants]].
+* Show an audit-style UI to explain to the user how frequently an [=origin=] synchronizes content through this API, with appropriate resource attribution.
+
+In addition, the user agent SHOULD:
+* Set [=effective minimum sync interval for origin=] in accordance with the user's engagement level with the [=origin=], as defined in [[#periodic-sync-scheduler-construct]]. In particular, this means the implementation stops firing [=periodicsync event|periodicsync events=] for [=periodic sync registrations=] whose associated [=origin=] the user has stopped engaging with.
+
```

Comment:
Would it help (and, is it possible) to require that only origins that the user has navigated to and engaged with in the *top-level frame* be able to use PBS?

Reply:
Could you explain what you're trying to fix with this requirement? Could you also clarify what you mean by using PBS?

We disallow registration if there's no top level or auxiliary frame open to the origin.

Reply:
Yep, and "engagement" isn't defined, but I think it's fair to assume engagement means "deliberate engagement", which means top level frames. We're not attempting to define engagement here, so I don't think we need to state the top-level thing beyond what we already do in `register()`.